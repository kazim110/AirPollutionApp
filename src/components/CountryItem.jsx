import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';

const CountryItem = ({ country }) => (
  <div className="col-6 country-item">
    <div className="d-flex justify-content-around">
      <span className="">{`Lat: ${country.latitude}`}</span>
      <span className="">{`Lon: ${country.longitude}`}</span>
      <NavLink className="country-data" to={`/stats?lat=${country.latitude}&lon=${country.longitude}`}>
        <BsArrowRightCircle className=" text-2xl" />
      </NavLink>
    </div>
    <div>
      <img src={`${country.image}`} alt={`${country.country}`} className="img-fluid" />
    </div>
    <div>
      <h3 className="">{country.country}</h3>
      <p className="">{country.capital}</p>
    </div>
  </div>
);

CountryItem.propTypes = {
  country: PropTypes.shape({
    country: PropTypes.string.isRequired,
    capital: PropTypes.string.isRequired,
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default CountryItem;
