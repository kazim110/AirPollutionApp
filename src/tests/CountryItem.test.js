import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CountryItem from '../components/CountryItem'; // Double-check the path to your component

// Describe your test
describe('CountryItem component', () => {
  // Test rendering the component with correct data
  it('should render correctly with correct data', () => {
    const component = renderer.create(
      <MemoryRouter>
        <CountryItem
          country={{
            latitude: 46.9479,
            longitude: 7.4474,
            image:
              'https://raw.githubusercontent.com/Ginohmk/worldMaps/main/maps/ch/vector.svg',
            country: 'Switzerland',
            capital: 'Bern',
          }}
        />
      </MemoryRouter>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
