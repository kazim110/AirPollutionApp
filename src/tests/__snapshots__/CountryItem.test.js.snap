// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryItem component should render correctly with correct data 1`] = `
<div
  className="col-6 country-item"
>
  <div
    className="d-flex p-1 justify-content-around"
  >
    <span
      className=""
    >
      Lat: 46.9479
    </span>
    <span
      className=""
    >
      Lon: 7.4474
    </span>
    <a
      className="country-data"
      href="/pollutionsDetails?lat=46.9479&lon=7.4474"
      onClick={[Function]}
    >
      <svg
        className="detailBtn"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
  <div
    className="p-3"
  >
    <img
      alt="Switzerland"
      className="img-fluid"
      src="https://raw.githubusercontent.com/Ginohmk/worldMaps/main/maps/ch/vector.svg"
    />
  </div>
  <div>
    <h3>
      Switzerland
    </h3>
    <p>
      Bern
    </p>
  </div>
</div>
`;
